### PR TITLE
UI doesn't look good in the pop-up xwikisas#9

### PR DESCRIPTION
**Pull request by oanalavinia**
* I replaced the old ConfirmationBox with a bootstrap modal for a better display

### CODE REVIEW

File: application-flashmessages-ui/src/main/resources/Flash/DisplayFlashMessages.xml [hunk 1]
```diff
@@ -183,14 +183,20 @@
     }
   #end
     ];
-    window.flashNotifier(params);
+    require(['jquery'], function($) {
+      $(window)[0].flashNotifier(params);
```
Comment (mflorea) [inline application-flashmessages-ui/src/main/resources/Flash/DisplayFlashMessages.xml]: I'm not sure what is the advantage of using ``$(window)[0]`` over simply ``window``. If there's no difference then I would keep the code simple.

File: application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml [hunk 2]
```diff
@@ -153,51 +153,64 @@
...
-      <code>/**
- * Confirm that the user has seen the popups
- */
-window.flashNotifierConfirm = function(params){
-  window.flashPopup.closeDialog();
-  new Ajax.Request("$doc.getURL('get')", {
-    parameters : {'ajax' : '1', 'flash': params['document']},
-    method: 'get'
+      <code>require(['jquery'], function($) {
+  var thisWindow = $(window)[0];
```
Comment (mflorea) [inline application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml]: Same as above.

File: application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml [hunk 3]
```diff
@@ -153,51 +153,64 @@
...
+    thisWindow.flashNotifierIndex = 0;
+    thisWindow.flashNotifierNext();
+  };
+  /**
+  * Display the next popup.
+  */
+  thisWindow.flashNotifierNext = function() {
+    // If index &gt; length, stop it!
+    if(thisWindow.flashNotifierIndex &gt;= thisWindow.flashNotifierParams.length)
+      return;
```
Comment (mflorea) [inline application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml]: It's not a good idea to rely only on indentation. The next person adding a line under this if might forget to add the ``{...}``.

File: application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml [hunk 4]
```diff
@@ -153,51 +153,64 @@
...
-  window.flashNotifierIndex = 0;
-  window.flashNotifierNext();
-};
-</code>
+  /**
+   * Events on confirmation.
+   */
+  $(document).on('click', '#flashPopup input.btn-primary', function(event) {
+    var params = thisWindow.flashNotifierParams[thisWindow.flashNotifierIndex];
+    // We need to wait for the displayed modal to be closed.
```
Comment (mflorea) [inline application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml]: I would rather listen to ``hidden.bs.modal`` and see if the event holds information about how the modal was closed (which button was clicked). If it doesn't then you can store this information yourself on ``click`` and use it later on ``hidden.bs.modal``.

**Comment by oanalavinia** [comment]
This is how the popup looks now
modalpopup

File: application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml [hunk 5]
```diff
@@ -153,51 +153,70 @@
...
+      window.flashPopup.modal('show');
+    }
+  }
+  /**
+   * Confirm that the user has seen the popups.
+   */
+  window.flashNotifierConfirm = function() {
+    let params = window.flashNotifierParams[window.flashNotifierIndex];
+    $.ajax({
+      url : "$doc.getURL('get')",
```
Comment (mflorea) [inline application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml]: You could use ``XWiki.currentDocument.getURL('get')`` to avoid the need to use Velocity.

File: application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml [hunk 6]
```diff
@@ -153,51 +153,70 @@
...
-  window.flashNotifierParams = params;
-  window.flashNotifierIndex = 0;
-  window.flashNotifierNext();
-};
-</code>
+  /**
+   * Save the fact that 'ok' button was clicked.
+   */
+  $(document).on('click', '#flashPopup input.btn-primary', function(event) {
+    $(this).data('clicked', 'yes');
```
Comment (mflorea) [inline application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml]: I would use a boolean value instead.

File: application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml [hunk 7]
```diff
@@ -153,51 +153,70 @@
...
+  });
+  /**
+   * Check if we can show the next popup when the modal is closed.
+   */
+  $(document).on('hidden.bs.modal', '#flashPopup', function(event) {
+    var clicked = $('#flashPopup input.btn-primary').data('clicked');
```
Comment (mflorea) [inline application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml]: You may need to reset the "clicked" value so that next time the modal is shown you don't think the button was clicked.

File: application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml (hunk 5 above)
Comment (oanalavinia) [reply application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml]: Because it's not always about the current document it's ok to use `new XWiki.Document('FlashNotifier', 'Flash').getURL('get')` ?

Comment (mflorea) [reply application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml]: Yes, for sure.
